src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -1624,6 +1647,9 @@ BlockIO InterpreterCreateQuery::doCreateOrReplaceTable(ASTCreateQuery & create,
         assert(done);
         created = true;
 
+        /// If table has dependencies - add them to the graph
+        addTableDependencies(create, query_ptr, getContext());
```

Comment:
Previously we didn't add dependencies for tables created with `CREATE OR REPLACE` when table didn't exist before.